Bug: rapid submissions in succession sporadically fail with 500 error

This problem was originally noted as:
`./prime test --run merge
`
 failing now sporadically with a 500 error, when run locally.  I think this may have something to do with the new metadata feature (see Exception in the comment below)?  It seems to occur most right after you restart the docker container?  Maybe some weird race condition with a transaction that is closing? 

See comment below for way to reproduce the problem.

Acceptance criteria:
1. Identify why this issue is happening.  Document the problem in this issue.
2. Fix the issue if it can be fixed.

FYI @MauriceReeves-usds

Here's what I see in the ReportStream logs when the error occurs:

```
prime_dev_1     | [2021-07-05T12:24:04.706Z] Writing deidentified report data to the DB
prime_dev_1     | [2021-07-05T12:24:04.713Z] Wrote 15 rows to test data table
prime_dev_1     | [2021-07-05T12:24:04.721Z] Function "reports" (Id: 10b50a3a-20ae-4904-9418-c51a1bad2fdb) invoked by Java Worker
prime_dev_1     | [2021-07-05T12:24:04.721Z] Executed 'Functions.reports' (Succeeded, Id=10b50a3a-20ae-4904-9418-c51a1bad2fdb, Duration=542ms)
prime_dev_1     | [2021-07-05T12:24:04.726Z] Executing 'Functions.reports' (Reason='This function was programmatically called via the host APIs.', Id=4307b38c-ddf4-4fbc-89d7-0bd2b8cb4d99)
prime_dev_1     | [2021-07-05T12:24:04.748Z] SQL [select "item".* from "public"."setting" as "item" join "public"."setting" as "org" on "item"."organization_id" = "org"."setting_id" where ("item"."is_active" = true and "item"."type" = ?::"public"."setting_type" and "item"."name" = ? and "org"."is_active" = true and "org"."type" = ?::"public"."setting_type" and "org"."organization_id" is null and "org"."name" = ?)]; ERROR: current transaction is aborted, commands ignored until end of transaction block
prime_dev_1     | [2021-07-05T12:24:04.748Z] Result: SQL [select "item".* from "public"."setting" as "item" join "public"."setting" as "org" on "item"."organization_id" = "org"."setting_id" where ("item"."is_active" = true and "item"."type" = ?::"public"."setting_type" and "item"."name" = ? and "org"."is_active" = true and "org"."type" = ?::"public"."setting_type" and "org"."organization_id" is null and "org"."name" = ?)]; ERROR: current transaction is aborted, commands ignored until end of transaction block
prime_dev_1     | [2021-07-05T12:24:04.748Z] Exception: SQL [select "item".* from "public"."setting" as "item" join "public"."setting" as "org" on "item"."organization_id" = "org"."setting_id" where ("item"."is_active" = true and "item"."type" = ?::"public"."setting_type" and "item"."name" = ? and "org"."is_active" = true and "org"."type" = ?::"public"."setting_type" and "org"."organization_id" is null and "org"."name" = ?)]; ERROR: current transaction is aborted, commands ignored until end of transaction block
prime_dev_1     | [2021-07-05T12:24:04.748Z] Stack: org.jooq.exception.DataAccessException: SQL [select "item".* from "public"."setting" as "item" join "public"."setting" as "org" on "item"."organization_id" = "org"."setting_id" where ("item"."is_active" = true and "item"."type" = ?::"public"."setting_type" and "item"."name" = ? and "org"."is_active" = true and "org"."type" = ?::"public"."setting_type" and "org"."organization_id" is null and "org"."name" = ?)]; ERROR: current transaction is aborted, commands ignored until end of transaction block
prime_dev_1     | [2021-07-05T12:24:04.748Z] 	at org.jooq_3.14.7.POSTGRES.debug(Unknown Source)
prime_dev_1     | [2021-07-05T12:24:04.748Z] 	at org.jooq.impl.Tools.translate(Tools.java:2880)
prime_dev_1     | [2021-07-05T12:24:04.748Z] 	at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:757)
prime_dev_1     | [2021-07-05T12:24:04.748Z] 	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:389)
prime_dev_1     | [2021-07-05T12:24:04.748Z] 	at org.jooq.impl.AbstractResultQuery.fetchLazy(AbstractResultQuery.java:481)
prime_dev_1     | [2021-07-05T12:24:04.748Z] 	at org.jooq.impl.AbstractResultQuery.fetchLazy(AbstractResultQuery.java:450)
prime_dev_1     | [2021-07-05T12:24:04.748Z] 	at org.jooq.impl.AbstractResultQuery.fetchLazyNonAutoClosing(AbstractResultQuery.java:464)
prime_dev_1     | [2021-07-05T12:24:04.748Z] 	at org.jooq.impl.AbstractResultQuery.fetchOne(AbstractResultQuery.java:641)
prime_dev_1     | [2021-07-05T12:24:04.748Z] 	at org.jooq.impl.SelectImpl.fetchOne(SelectImpl.java:3074)
prime_dev_1     | [2021-07-05T12:24:04.748Z] 	at gov.cdc.prime.router.azure.DatabaseAccess.fetchSetting(DatabaseAccess.kt:309)
prime_dev_1     | [2021-07-05T12:24:04.749Z] 	at gov.cdc.prime.router.azure.SettingsFacade$findSetting$setting$1.invoke(SettingsFacade.kt:92)
prime_dev_1     | [2021-07-05T12:24:04.749Z] 	at gov.cdc.prime.router.azure.SettingsFacade$findSetting$setting$1.invoke(SettingsFacade.kt:89)
prime_dev_1     | [2021-07-05T12:24:04.749Z] 	at gov.cdc.prime.router.azure.DatabaseAccess.transactReturning$lambda-1(DatabaseAccess.kt:69)
prime_dev_1     | [2021-07-05T12:24:04.749Z] 	at org.jooq.impl.DefaultDSLContext.lambda$transactionResult0$0(DefaultDSLContext.java:543)
prime_dev_1     | [2021-07-05T12:24:04.749Z] 	at org.jooq.impl.Tools$35$1.block(Tools.java:5203)
prime_dev_1     | [2021-07-05T12:24:04.749Z] 	at java.base/java.util.concurrent.ForkJoinPool.managedBlock(ForkJoinPool.java:3128)
prime_dev_1     | [2021-07-05T12:24:04.749Z] 	at org.jooq.impl.Tools$35.get(Tools.java:5200)
prime_dev_1     | [2021-07-05T12:24:04.749Z] 	at org.jooq.impl.DefaultDSLContext.transactionResult0(DefaultDSLContext.java:595)
prime_dev_1     | [2021-07-05T12:24:04.749Z] 	at org.jooq.impl.DefaultDSLContext.transactionResult(DefaultDSLContext.java:512)
prime_dev_1     | [2021-07-05T12:24:04.749Z] 	at gov.cdc.prime.router.azure.DatabaseAccess.transactReturning(DatabaseAccess.kt:69)
prime_dev_1     | [2021-07-05T12:24:04.749Z] 	at gov.cdc.prime.router.azure.SettingsFacade.findSetting(SettingsFacade.kt:89)
prime_dev_1     | [2021-07-05T12:24:04.749Z] 	at gov.cdc.prime.router.azure.SettingsFacade.findSender(SettingsFacade.kt:64)
prime_dev_1     | [2021-07-05T12:24:04.749Z] 	at gov.cdc.prime.router.azure.ReportFunction.validateRequest(ReportFunction.kt:198)
prime_dev_1     | [2021-07-05T12:24:04.749Z] 	at gov.cdc.prime.router.azure.ReportFunction.run(ReportFunction.kt:78)
prime_dev_1     | [2021-07-05T12:24:04.749Z] 	at jdk.internal.reflect.GeneratedMethodAccessor546.invoke(Unknown Source)
prime_dev_1     | [2021-07-05T12:24:04.749Z] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
prime_dev_1     | [2021-07-05T12:24:04.749Z] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
prime_dev_1     | [2021-07-05T12:24:04.749Z] 	at com.microsoft.azure.functions.worker.broker.JavaMethodInvokeInfo.invoke(JavaMethodInvokeInfo.java:22)
prime_dev_1     | [2021-07-05T12:24:04.749Z] 	at com.microsoft.azure.functions.worker.broker.EnhancedJavaMethodExecutorImpl.execute(EnhancedJavaMethodExecutorImpl.java:55)
prime_dev_1     | [2021-07-05T12:24:04.749Z] 	at com.microsoft.azure.functions.worker.broker.JavaFunctionBroker.invokeMethod(JavaFunctionBroker.java:57)
prime_dev_1     | [2021-07-05T12:24:04.750Z] 	at com.microsoft.azure.functions.worker.handler.InvocationRequestHandler.execute(InvocationRequestHandler.java:33)
prime_dev_1     | [2021-07-05T12:24:04.750Z] 	at com.microsoft.azure.functions.worker.handler.InvocationRequestHandler.execute(InvocationRequestHandler.java:10)
prime_dev_1     | [2021-07-05T12:24:04.750Z] 	at com.microsoft.azure.functions.worker.handler.MessageHandler.handle(MessageHandler.java:45)
prime_dev_1     | [2021-07-05T12:24:04.750Z] 	at com.microsoft.azure.functions.worker.JavaWorkerClient$StreamingMessagePeer.lambda$onNext$0(JavaWorkerClient.java:92)
prime_dev_1     | [2021-07-05T12:24:04.750Z] 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
prime_dev_1     | [2021-07-05T12:24:04.750Z] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
prime_dev_1     | [2021-07-05T12:24:04.750Z] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
prime_dev_1     | [2021-07-05T12:24:04.750Z] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
prime_dev_1     | [2021-07-05T12:24:04.750Z] 	at java.base/java.lang.Thread.run(Thread.java:829)
prime_dev_1     | [2021-07-05T12:24:04.750Z] Caused by: org.postgresql.util.PSQLException: ERROR: current transaction is aborted, commands ignored until end of transaction block
prime_dev_1     | [2021-07-05T12:24:04.750Z] 	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2553)
prime_dev_1     | [2021-07-05T12:24:04.750Z] 	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2285)
prime_dev_1     | [2021-07-05T12:24:04.750Z] 	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:323)
prime_dev_1     | [2021-07-05T12:24:04.750Z] 	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:481)
prime_dev_1     | [2021-07-05T12:24:04.750Z] 	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:401)
prime_dev_1     | [2021-07-05T12:24:04.750Z] 	at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:164)
prime_dev_1     | [2021-07-05T12:24:04.751Z] 	at org.postgresql.jdbc.PgPreparedStatement.execute(PgPreparedStatement.java:153)
prime_dev_1     | [2021-07-05T12:24:04.751Z] 	at com.zaxxer.hikari.pool.ProxyPreparedStatement.execute(ProxyPreparedStatement.java:44)
prime_dev_1     | [2021-07-05T12:24:04.751Z] 	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.execute(HikariProxyPreparedStatement.java)
prime_dev_1     | [2021-07-05T12:24:04.751Z] 	at org.jooq.tools.jdbc.DefaultPreparedStatement.execute(DefaultPreparedStatement.java:214)
prime_dev_1     | [2021-07-05T12:24:04.751Z] 	at org.jooq.impl.Tools.executeStatementAndGetFirstResultSet(Tools.java:4178)
prime_dev_1     | [2021-07-05T12:24:04.751Z] 	at org.jooq.impl.AbstractResultQuery.execute(AbstractResultQuery.java:279)
prime_dev_1     | [2021-07-05T12:24:04.751Z] 	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:375)
prime_dev_1     | [2021-07-05T12:24:04.751Z] 	... 35 more
prime_dev_1     | [2021-07-05T12:24:04.751Z] Caused by: org.postgresql.util.PSQLException: ERROR: canceling statement due to user request
prime_dev_1     | [2021-07-05T12:24:04.751Z] 	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2553)
prime_dev_1     | [2021-07-05T12:24:04.751Z] 	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2285)
prime_dev_1     | [2021-07-05T12:24:04.751Z] 	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:323)
prime_dev_1     | [2021-07-05T12:24:04.751Z] 	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:481)
prime_dev_1     | [2021-07-05T12:24:04.751Z] 	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:401)
prime_dev_1     | [2021-07-05T12:24:04.752Z] 	at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:164)
prime_dev_1     | [2021-07-05T12:24:04.752Z] 	at org.postgresql.jdbc.PgPreparedStatement.execute(PgPreparedStatement.java:153)
prime_dev_1     | [2021-07-05T12:24:04.752Z] 	at com.zaxxer.hikari.pool.ProxyPreparedStatement.execute(ProxyPreparedStatement.java:44)
prime_dev_1     | [2021-07-05T12:24:04.752Z] 	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.execute(HikariProxyPreparedStatement.java)
prime_dev_1     | [2021-07-05T12:24:04.752Z] 	at org.jooq.tools.jdbc.DefaultPreparedStatement.execute(DefaultPreparedStatement.java:214)
prime_dev_1     | [2021-07-05T12:24:04.752Z] 	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:458)
prime_dev_1     | [2021-07-05T12:24:04.752Z] 	at org.jooq.impl.AbstractDMLQuery.execute(AbstractDMLQuery.java:947)
prime_dev_1     | [2021-07-05T12:24:04.752Z] 	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:375)
prime_dev_1     | [2021-07-05T12:24:04.752Z] 	at org.jooq.impl.Tools$35$1.block(Tools.java:5203)
prime_dev_1     | [2021-07-05T12:24:04.752Z] 	at java.base/java.util.concurrent.ForkJoinPool.managedBlock(ForkJoinPool.java:3118)
prime_dev_1     | [2021-07-05T12:24:04.752Z] 	at org.jooq.impl.Tools$35.get(Tools.java:5200)
prime_dev_1     | [2021-07-05T12:24:04.752Z] 	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
prime_dev_1     | [2021-07-05T12:24:04.752Z] 	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1692)
prime_dev_1     | [2021-07-05T12:24:04.752Z] 	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
prime_dev_1     | [2021-07-05T12:24:04.752Z] 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
prime_dev_1     | [2021-07-05T12:24:04.752Z] 	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
prime_dev_1     | [2021-07-05T12:24:04.753Z] 	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
prime_dev_1     | [2021-07-05T12:24:04.753Z] 	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)

```

--- a/prime-router/docker-compose.yml
+++ b/prime-router/docker-compose.yml
@@ -72,7 +72,7 @@ services:
 
   settings:
     build: settings/.
-    command: "--wait=25 --check-last-modified prime_dev /settings/organizations-local.yml"
+    command: "--wait=20 --check-last-modified prime_dev /settings/organizations-local.yml"
     depends_on:
       - prime_dev
     volumes:

--- a/prime-router/docker-compose.yml
+++ b/prime-router/docker-compose.yml
@@ -72,7 +72,7 @@ services:
 
   settings:
     build: settings/.
-    command: "--wait=15 --check-last-modified prime_dev /settings/organizations-local.yml"
+    command: "--wait=25 --check-last-modified prime_dev /settings/organizations-local.yml"
     depends_on:
       - prime_dev
     volumes:

--- a/prime-router/src/main/kotlin/azure/BlobAccess.kt
+++ b/prime-router/src/main/kotlin/azure/BlobAccess.kt
@@ -4,6 +4,8 @@ import com.azure.storage.blob.BlobClient
 import com.azure.storage.blob.BlobClientBuilder
 import com.azure.storage.blob.BlobContainerClient
 import com.azure.storage.blob.BlobServiceClientBuilder
+import com.azure.storage.blob.models.BlobErrorCode
+import com.azure.storage.blob.models.BlobStorageException
 import gov.cdc.prime.router.Report
 import gov.cdc.prime.router.serializers.CsvSerializer
 import gov.cdc.prime.router.serializers.Hl7Serializer
@@ -107,7 +109,16 @@ class BlobAccess(
         val blobConnection = System.getenv(blobConnEnvVar)
         val blobServiceClient = BlobServiceClientBuilder().connectionString(blobConnection).buildClient()
         val containerClient = blobServiceClient.getBlobContainerClient(name)
-        if (!containerClient.exists()) containerClient.create()
+        try {
+            if (!containerClient.exists()) containerClient.create()
+        } catch (error: BlobStorageException) {
+            // This can happen when there are concurrent calls to the API
+            if (error.errorCode.equals(BlobErrorCode.CONTAINER_ALREADY_EXISTS)) {
+                logger.warn("Container $name already exists")
+            } else {
+                throw error
+            }
+        }
         return containerClient
     }
 

